src/index.ts

```diff
@@ -418,11 +423,26 @@ export const newForm: NewFormFn = <T extends object>(
       [field]: value,
     }));
 
+    const isDirtyNow = checkIsDirty();
+    __isDirty.set(isDirtyNow);
```

Comment:
I think you can use a single line for this

```suggestion
    __isDirty.set(checkIsDirty());
```